File: Python/players.py
from utils import user_move, random_move, minmax_move


class Player:
    def __init__(self, name):
        self.name = name

    def __repr__(self):
        return self.name

    def makeMove(self, game, mode):
        if mode==0:
            moveId = random_move(game, self.name)
        elif mode==1:
            moveId = user_move(game, self.name)
        elif mode==2:
            moveId = minmax_move(game, self.name, alpha_beta=0)
        else:
            moveId = minmax_move(game, self.name, alpha_beta=1)
        return moveId


class HumanPlayer(Player):
    def __init__(self, name="o"):
        super().__init__(name)


class AIPlayer(Player):
    def __init__(self, name="x"):
        super().__init__(name)

File: Javascript/app/js/Players.js
import { minmax_move, random_move } from "./utils.js";

class Player {
  constructor(name) {
    this.name = name;
  }

  makeMove(game, mode) {
    return;
  }
}

class HumanPlayer extends Player {
  constructor(name = "o") {
    super(name);
  }

  makeMove(game, mode) {
    return;
  }
}

class AIPlayer extends Player {
  constructor(name = "x") {
    super(name);
  }

  makeMove(game, mode) {
    // 0: Random move
    // 1: User input
    // 2: Minmax algorithm
    // 3: Minmax Alpha Beta algorithm
    console.log("Mode : ", mode);
    if (mode == 0) {
      return random_move(game, this.name);
    } else if (mode == 1) {
      return random_move(game, this.name);
    } else if (mode == 2) {
      return minmax_move(game, this.name, 0);
    } else {
      return minmax_move(game, this.name, 1);
    }
  }
}

export { HumanPlayer, AIPlayer };
